pilot/pkg/bootstrap/server.go

```diff
@@ -955,9 +964,21 @@ func (s *Server) initDiscoveryService(args *PilotArgs) error {
 	}
 	s.mux = discovery.RestContainer.ServeMux
 
+	if model.JwtKeyResolver == nil {
+		log.Info("Creating a JwtKeyResolver for the first time")
+		// Public key refresh interval must be a positive number.
+		var jwtPubKeyRefreshInterval time.Duration
+		if jwtPubKeyRefreshInterval = args.JWKSOptions.JWTPubKeyRefreshInterval; jwtPubKeyRefreshInterval == 0 {
```

Comment:
Can we hide all these under NewJwksResolver function instead.

Reply:
Done!